homeassistant/helpers/translation.py

```diff
@@ -263,17 +264,20 @@ async def _async_load(self, language: str, components: set[str]) -> None:
         _LOGGER.debug(
             "Cache miss for %s: %s",
             language,
-            ", ".join(components),
+            components,
         )
         # Fetch the English resources, as a fallback for missing keys
         languages = [LOCALE_EN] if language == LOCALE_EN else [LOCALE_EN, language]
 
         integrations: dict[str, Integration] = {}
-        domains = list({loaded.partition(".")[0] for loaded in components})
+        domains = {loaded.partition(".")[0] for loaded in components}
         ints_or_excs = await async_get_integrations(self.hass, domains)
         for domain, int_or_exc in ints_or_excs.items():
             if isinstance(int_or_exc, Exception):
-                raise int_or_exc
+                _LOGGER.warning(
+                    "Failed to load integration for translation: %s", int_or_exc
+                )
+                continue
```

Comment:
It somehow feels like it would be nice to keep this change of behavior outside this optimization PR. Feel free to move it to a separate PR or not. Regardless of which, there should be a test which tests for the new behavior and fails with the old behavior.

Reply:
Added coverage for partial failure 85327a1251509923065d0ac3ed97482c2f71dd1d